packages/react-components/react-table/src/components/TableHeaderCell/TableHeaderCell.types.ts

```diff
@@ -10,7 +10,8 @@ export type TableHeaderCellSlots = {
   /**
    * Button handles correct narration and interactions for sorting;
    */
-  button: NonNullable<Slot<ARIAButtonSlotProps>>;
+  button: NonNullable<ARIAButtonSlotProps>;
```

Comment:
`button` shouldn't be supported at all, but now I don't want to break API. if users want to explicitly render a `button` they can deal with the a11y consequences. We've made the happy path accessible